pagopa/io-app
# fix(Redesign Servizi): [IARS-43] Render maximum amount of information available

## Short description
When a service's data cannot be fetched due to any error, the app should render as much information as it has already. Examples are contact preferences and fiscal code.

~Please note that I couldn't write a proper test for this bug due to an issue with the `navigation` parameter injection. We'll see how to address it in the future but I think it goes beyond the scope of this task.~ Thanks to @debiff for providing guidance, I was able to write [a first test](https://github.com/pagopa/io-app/pull/3268/files#diff-ab11534b5ccaf3963833373014cba7c3edd83bd0bbf7f0e758e58d22218c9ca1R30).

## List of changes proposed in this pull request
- Make `servicesMetada` optional in [ServiceMetadata](ts/components/services/ServiceMetadata/index.tsx) to always render at least fiscal code and ID
- Simplify rendering logic based on [simple flags](https://github.com/pagopa/io-app/compare/IARS-43-show-persisted-service-data?expand=1#diff-18e4ee1a39e556a9f274cc9b9ac60284fa9159d6584025d8b13d88a60aa9e6bdR162)
- render immediately when an error is present 
- I removed [unused props](https://github.com/pagopa/io-app/compare/IARS-43-show-persisted-service-data?expand=1#diff-18e4ee1a39e556a9f274cc9b9ac60284fa9159d6584025d8b13d88a60aa9e6bdL300), [redundant checks](https://github.com/pagopa/io-app/compare/IARS-43-show-persisted-service-data?expand=1#diff-18e4ee1a39e556a9f274cc9b9ac60284fa9159d6584025d8b13d88a60aa9e6bdL114) and [tests](ts/screens/services/__tests__/ServiceDetailsScreen.test.ts).

## How to test
1. configure the [dev server endpoint](https://github.com/pagopa/io-dev-api-server/blob/18230a125801868464e215c90664f076834746f6/src/routers/services_metadata.ts#L26) to return an error like:
```typescript
res.status(500).send({ message: "wat?" })
```
2. navigate to the services page, select LOCAL ones
3. open the first service
4. note that contact preferences and fiscal code are rendered immediately


| Before | After |
|--------|------|
| ![Screenshot 2021-07-29 at 17 58 32](https://user-images.githubusercontent.com/2094604/127525389-203631ad-a210-4647-96bb-66227d3b6d0f.png) | ![Screenshot 2021-07-29 at 16 11 22](https://user-images.githubusercontent.com/2094604/127525063-a3871ca8-f674-4c9c-b2f3-72343c124713.png) |

Screencast showing the two scenarios: `service0` fails while `service1` succeeds. This way you can compare the loading logic to the existing one.


https://user-images.githubusercontent.com/2094604/127526607-e21e0ec5-5a5f-4aa2-9b47-89d3ab8a9aab.mov
